### PR TITLE
Update Python SDK for receipts and events

### PR DESCRIPTION
**Pull request by aludvik**
This depends on #844

### CODE REVIEW

File: sdk/python/sawtooth_sdk/processor/state.py [hunk 1]
```diff
@@ -112,3 +113,47 @@ def delete(self, addresses, timeout=None):
...
+        response = state_context_pb2.TpAddReceiptDataResponse()
+        response.ParseFromString(
+            self._stream.send(
+                Message.TP_ADD_RECEIPT_DATA_REQUEST,
+                request).result(timeout).content)
+        if response.status == state_context_pb2.TpAddReceiptDataResponse.ERROR:
+            raise InvalidTransaction(
+                "Failed to add receipt data: {}".format((data_type, data)))
+
+    def add_event(self, event_type, attributes=None, data=None, timeout=None):
```
Comment (agunde406) [inline sdk/python/sawtooth_sdk/processor/state.py]: Docstring please!

**Comment by agunde406** [comment]
I also suggest removing the rename of State->Contexts from this PR as there is probably a significant amount of documentation that would need to be reviewed and updated.

**Comment by aludvik** [comment]
Added docstring to `add_event`. Passed on other commits and doc updating work to @RyanLassigBanks

**Comment by aludvik** [comment]
Added tests

File: sdk/python/sawtooth_sdk/processor/state.py [hunk 2]
```diff
@@ -112,3 +113,59 @@ def delete(self, addresses, timeout=None):
...
+        """Add a new event to the execution result for this transaction.
+
+        Args:
+            event_type (str): This is used to subscribe to events. It should be
+                globally unique and describe what, general, has occured.
```
Comment (boydjohnson) [inline sdk/python/sawtooth_sdk/processor/state.py]: general -> in general ?

**Pull request merged**